cpu/stm32/dist/irqs/gen_vectors.py

```diff
@@ -0,0 +1,143 @@
+#!/usr/bin/env python3
+
+# Copyright (C) 2020 Inria
+#
+# This file is subject to the terms and conditions of the GNU Lesser
+# General Public License v2.1. See the file LICENSE in the top level
+# directory for more details.
+
+"""Generate the vectors table for a given STM32 CPU line."""
+
+import os
+import argparse
+import re
+
+
+CURRENT_DIR = os.path.dirname(os.path.abspath(__file__))
+RIOTBASE = os.getenv(
+    "RIOTBASE", os.path.abspath(os.path.join(CURRENT_DIR, "../../../..")))
+STM32_VECTORS_DIR = os.path.join(RIOTBASE, "cpu/stm32/vectors")
+STM32_VENDOR_DIR = os.path.join(RIOTBASE, "cpu/stm32/include/vendor")
+STM32_CMSIS_FILE = os.path.join(
+    RIOTBASE, STM32_VENDOR_DIR, "cmsis/{}/Include/{}.h")
+
+VECTORS_FORMAT = """
+/*
+ * PLEASE DON'T EDIT
+ *
+ * This file was automatically generated by
+ * ./cpu/stm32/dist/irqs/gen_vectors.py
+ */
+
+#include "vectors_cortexm.h"
+
+/* define a local dummy handler as it needs to be in the same compilation unit
+ * as the alias definition */
+void dummy_handler(void) {{
+    dummy_handler_default();
+}}
+
+/* {cpu_line} specific interrupt vectors */
+{isr_lines}
+
+/* CPU specific interrupt vector table */
+ISR_VECTOR(1) const isr_t vector_cpu[CPU_IRQ_NUMOF] = {{
+{irq_lines}
+}};
+"""
```

Comment:
Better put this in a file?

Reply:
see https://github.com/RIOT-OS/RIOT/pull/14595#discussion_r481214897